future-apps/payroll/contracts/Payroll.sol

```diff
@@ -686,17 +686,16 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
     /**
      * @dev Try removing employee if there are no pending payments and has reached employee's end date.
      * @param _employeeId Employee's identifier
-     * @param _when Date timestamp used to evaluate if the employee can be removed from the payroll.
      */
-    function _tryRemovingEmployee(uint256 _employeeId, uint64 _when) private {
+    function _tryRemovingEmployee(uint256 _employeeId) private {
         Employee storage employee = employees[_employeeId];
 
-        bool hasReachedEndDate = employee.endDate <= _when;
-        bool isOwedSalary = _getOwedSalary(_employeeId, _when) > 0;
-        bool isOwedAccruedValue = employee.accruedValue > 0;
-        bool areNoPendingPayments = !isOwedSalary && !isOwedAccruedValue;
+        bool hasReachedEndDate = employee.endDate <= getTimestamp64();
+        bool noPendingSalary = _getOwedSalary(_employeeId) == 0;
+        bool noPendingAccruedValue = employee.accruedValue == 0;
+        bool noPendingPayments = noPendingSalary && noPendingAccruedValue;
 
-        if (hasReachedEndDate && areNoPendingPayments) {
+        if (hasReachedEndDate && noPendingPayments) {
             delete employeeIds[employee.accountAddress];
             delete employees[_employeeId];
         }
```

Comment:
I would still suggest using shortcuts, as explained [here](https://github.com/aragon/aragon-apps/pull/736#discussion_r269501064) to save some gas in the majority of the uses of this function.
```
if (employee.endDate > getTimestamp64()) {
    return;
}
if (_getOwedSalary(_employeeId) > 0) {
   return;
}
if (employee.accruedValue > 0) {
   return;
}

delete employeeIds[employee.accountAddress];
delete employees[_employeeId];
```
Don't you think?

Reply:
Completely, I still think that boolean algebra was one of the easiest subjects I learned and one of those that I underestimated most at the same time 😂

Reply:
Hahaha, so true!